src/main/java/net/atos/zac/app/mailtemplate/MailtemplateRESTService.java

```diff
@@ -28,13 +36,28 @@ public class MailtemplateRESTService {
     @Inject
     private RESTMailtemplateConverter restMailtemplateConverter;
 
+    @Inject
+    private ZaakafhandelParameterService zaakafhandelParameterService;
+
+    @Inject
+    private ZRCClientService zrcClientService;
+
     @GET
     @Path("{mailtemplateEnum}/{zaakUUID}")
     public RESTMailtemplate findMailtemplate(@PathParam("mailtemplateEnum") final Mail mail,
             @PathParam("zaakUUID") final UUID zaakUUID) {
-        // TODO Controleren adhv het zaakUUID of er een ander mailtemplate dan de default is ingesteld op het zaaktype.
-        return mailTemplateService.findMailtemplate(mail)
-                .map(restMailtemplateConverter::convert)
-                .orElse(null);
+        final Zaak zaak = zrcClientService.readZaak(zaakUUID);
+        final ZaakafhandelParameters zaakafhandelParameters =
+                zaakafhandelParameterService.readZaakafhandelParameters(UriUtil.uuidFromURI(zaak.getZaaktype()));
+        final Optional<MailtemplateKoppeling> mailtemplateKoppeling =
```

Comment:
En dan nu functioneel (onderstande is een vervanging voor regel 52 tot einde methode): 

        return zaakafhandelParameters.getMailtemplateKoppelingen().stream()
                .filter(koppeling -> koppeling.getMailTemplate().getMail().equals(mail))
                .map(koppeling -> restMailtemplateConverter.convert(koppeling.getMailTemplate()))
                .findFirst()
                .orElse(mailTemplateService.findDefaultMailtemplate(mail)
                                .map(restMailtemplateConverter::convert)
                                .orElse(null));

Je riep in jouw code de methode mailTemplateService.readMailtemplate(mail) aan en controleerde daarna op null. 
Een readXXX()  methode geeft nooit een null terug maar gooit in dat geval een exceptie. Je moet dan dus de findXXX() gebruiken.
Ik zie hier dat de findXXX() methode alleen niet dezelfde naam heeft dan de readXXX() variant. 
Geef ze allebei even dezelfde naam

Reply:
aangepast